### PR TITLE
Make PageFaultsInfo a struct rather than a Timer

### PR DESCRIPTION
**Pull request by florian-kuebler**
This follows #4417 for the page fault info.

Test: Take a capture with page fault info.
Bug: http://b/192335025

### CODE REVIEW

File: src/ClientData/include/ClientData/PageFaultsInfo.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef CLIENT_DATA_PAGE_FAULTS_INFO_H_
```
Comment (dpallotti) [inline src/ClientData/include/ClientData/PageFaultsInfo.h:4]:
```suggestion
// found in the LICENSE file.

#ifndef CLIENT_DATA_PAGE_FAULTS_INFO_H_
```

File: src/ClientData/include/ClientData/PageFaultsInfo.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  [[nodiscard]] bool HasMinorPageFaultsInfo() const {
+    return system_page_faults != kMissingInfo && system_major_page_faults != kMissingInfo &&
+           cgroup_page_faults != kMissingInfo && cgroup_major_page_faults != kMissingInfo &&
+           process_minor_page_faults != kMissingInfo;
+  }
+
+ private:
+  static constexpr int64_t kMissingInfo = -1;
```
Comment (dpallotti) [inline src/ClientData/include/ClientData/PageFaultsInfo.h:33]: For assignment, the value in `GrpcProtos` is used, right? So now I'm wondering if having this separately makes sense, when we probably can't get rid of the one in `GrpcProtos`.

**Review by dpallotti — APPROVED** [review]
I'm still wondering about the `kMissingInfo` now that I think I realized the assignment of that value happens on the service and has to use the constant in `GrpcProtos`.

File: src/ClientData/include/ClientData/PageFaultsInfo.h (hunk 1 above)
Comment (florian-kuebler) [reply src/ClientData/include/ClientData/PageFaultsInfo.h]: Done.

File: src/ClientData/include/ClientData/PageFaultsInfo.h (hunk 2 above)
Comment (florian-kuebler) [reply src/ClientData/include/ClientData/PageFaultsInfo.h]: Agree. Changed it at both places.

**Pull request merged**